### PR TITLE
Simplify media insertion flow

### LINKED ISSUE #946: Simplify media insertion flow
## Problem

As of right now, the process of adding an image block (and soon additional types of media) is a bit cumbersome. I'd like to make improvements to the media selection part of the flow, but for now at very least we can simplify by skipping the placeholder step or combining it with the location options sheet.

_Note: I'm not sure if any additional design is needed, but if so I can pull together a blueprint._

## Current Flow

image

1. Tap `(+)` (inserter) – this shows the Block Library sheet
2. Tap `Image` to add Image block placeholder to canvas
3. Tap on block placeholder to trigger sheet
4. Tap on an option to choose location
5. Select image to upload
6. Image gets added to canvas, uploads, succeeds/fails

## Proposed Flow

image

1. Tap `(+)` (inserter) – this shows the Block Library sheet
2. Tap `Image` to add Image block placeholder to canvas –  _trigger media location sheet_. _Note: the placeholder could still display on the canvas, behind the sheet_, and be dismissed if the user backs out of the process by tapping `Cancel`.
3. Choose location
4. Select image
5. Image added to canvas, uploads, succeeds/fails

## Note on Future Improvements

I'd like to replace the location-choosing sheet with an improved media selection sheet, similar to Aztec – but for now, let's at least make this improvement as it will benefit some of the next blocks we're building (Video, for example).

### PR DESCRIPTION
**Pull request by jd-alexander**
Fixes #946
**Gutenberg PR** https://github.com/WordPress/gutenberg/pull/29546

This change simplifies the media insertion flow by showing the media options sheet when you tap the Media block to add the Media placeholder to the canvas. Before you would have to add the placeholder and click for eg. Add Image for it to be shown.
To test:
https://github.com/WordPress/gutenberg/pull/29546
PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/docs/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

**Review by guarani — APPROVED** [review]
Just adding my approval here, since I already reviewed and approved https://github.com/WordPress/gutenberg/pull/25031#pullrequestreview-510582123.

**Review by guarani — APPROVED** [review]
Just approving this since the linked Gutenberg PRs are approved. I think this is missing an entry in the release notes though (and we're missing an entry in the `react-native-editor` package change log).

**Comment by jd-alexander** [comment]
> Just approving this since the linked Gutenberg PRs are approved. I think this is missing an entry in the release notes though (and we're missing an entry in the `react-native-editor` package change log).

Thanks. I was planning to put the release notes in and request a review from you. Can I just put them in and merge afterwards ?

**Comment by guarani** [comment]
> > Just approving this since the linked Gutenberg PRs are approved. I think this is missing an entry in the release notes though (and we're missing an entry in the `react-native-editor` package change log).
> 
> Thanks. I was planning to put the release notes in and request a review from you. Can I just put them in and merge afterwards ?

Yeah sure, I don't mind either way.

**Comment by jd-alexander** [comment]
> > > Just approving this since the linked Gutenberg PRs are approved. I think this is missing an entry in the release notes though (and we're missing an entry in the `react-native-editor` package change log).
> 
> > 
> 
> > Thanks. I was planning to put the release notes in and request a review from you. Can I just put them in and merge afterwards ?
> 
> 
> 
> Yeah sure, I don't mind either way.

No problem. I will sync with trunk and get all these PRs merged in on Monday.

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -22,6 +22,7 @@ Unreleased
 1.49.0
 ------
 * [*] Remove the cancel button from settings options (Android only) [https://github.com/WordPress/gutenberg/pull/29599]
+* [**] The media upload options of the Image, Video and Gallery block automatically opens when the respective block is inserted. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/2700]
```
Comment (fluiddot) [inline RELEASE-NOTES.txt]: I'd move this line to `Unreleased` section since this changes haven't released yet.

Comment (jd-alexander) [reply RELEASE-NOTES.txt]: Good catch. Thanks! Seems merging in develop changed things up because it was under unreleased initially 7b8dcaf 😅

**Pull request merged**